animation_nodes/base_types/tree_auto_execution.py

```diff
@@ -16,69 +14,96 @@ class AutoExecutionTrigger_MonitorProperty(bpy.types.PropertyGroup):
         items = idTypeItems)
 
     object: PointerProperty(type = bpy.types.Object, name = "Object")
+    collection: PointerProperty(type = bpy.types.Collection, name = "Collection")
     scene: PointerProperty(type = bpy.types.Scene, name = "Scene")
-    dataPath: StringProperty(name = "Data Path", default = "")
+    dataPaths: StringProperty(name = "Data Paths", default = "",
+        description = "Comma separated paths of properties to monitor")
 
     lastState: StringProperty(default = "")
+    expanded: BoolProperty(default = True)
     enabled: BoolProperty(default = True)
     hasError: BoolProperty(default = False)
 
     def update(self):
         lastState = self.lastState
-        newState = self.getPropertyState()
+        newState = self.getPropertiesState()
         if newState == lastState:
             return False
         else:
             self.lastState = newState
             return self.enabled and lastState != ""
 
-    def getPropertyState(self):
-        prop = self.getProperty()
-        if prop is None: return ""
-        if hasattr(prop, "__iter__"):
-            return " ".join(str(part) for part in prop)
-        return str(prop)
+    def getPropertiesState(self):
+        props = self.getProperties()
+        if len(props) == 0: return ""
+
+        propsString = ""
+        for prop in props:
+            if hasattr(prop, "__iter__"):
+                propsString += "".join(str(part) for part in prop)
+            else: propsString += str(prop)
 
-    def getProperty(self):
+        return propsString
+
+    def getProperties(self):
         self.hasError = False
         object = self.getObject()
-        if object is None or self.dataPath is "":
-            return None
-
-        try: return object.path_resolve(self.dataPath)
+        if object is None or self.dataPaths.strip() is "":
+            return []
+
+        try:
+            paths = self.dataPaths.split(",")
+            if self.idType == "COLLECTION":
+                properties = []
+                for obj in object.all_objects:
+                    properties.extend(obj.path_resolve(p.strip()) for p in paths)
+                return properties
+            else:
+                return [object.path_resolve(p.strip()) for p in paths]
         except:
             self.hasError = True
-            return None
+            return []
 
     def getObject(self):
```

Comment:
Maybe change this function so that it returns a list of ID blocks.
Then call does not need to have a special case for collections.